client/my-sites/checkout/index.js

```diff
@@ -27,11 +27,12 @@ import {
 export default function () {
 	page( '/checkout*', recordSiftScienceUser );
 
+	// Jetpack siteless checkout works logged-out, so do not include redirectLoggedOut or siteSelection.
 	page(
 		`/checkout/jetpack/:productSlug`,
 		setLocaleMiddleware(),
 		noSite,
-		checkoutSiteless,
+		checkoutJetpackSiteless,
```

Comment:
This PR renames this controller from `checkoutSiteless` to `checkoutJetpackSiteless` to clarify the difference between this route and other siteless checkout flows.